ibm_cloud_sdk_core/api_exception.py

```diff
@@ -39,15 +40,28 @@ def __init__(self, code: int, *, message: Optional[str] = None, http_response: O
         # Call the base class constructor with the parameters it needs
         super().__init__(message)
         self.message = message
-        self.code = code
+        self.status_code = code
         self.http_response = http_response
         self.global_transaction_id = None
         if http_response is not None:
             self.global_transaction_id = http_response.headers.get('X-Global-Transaction-ID')
             self.message = self.message if self.message else self._get_error_message(http_response)
 
+    # pylint: disable=fixme
+    # TODO: delete this by the end of 2024.
+    @property
+    def code(self):
+        """The old `code` property with a deprecation warning."""
+
+        warnings.warn(
+            'Using the `code` attribute on the `ApiException` is deprecated and'
+            'will be removed in the future. Use `status_code` instead.',
+            DeprecationWarning,
+        )
+        return self.status_code
+
```

Comment:
I thought it would be good to not remove the old property completely, but mark it deprecated and give some time to the the users to update their code. Also, the date in the comment is arbitrary, there is no real reason behind it. :)

Reply:
By renaming the field from `code` to `status_code` above, I think you've introduced a breaking change despite the addition of this new getter function.
What I mean is...  previously a user would need to access the `code` field like this:
```
    print('The status code is: ', exception.code)
```
That will no longer work as the user would need to do one of the following instead now:
```
    print('The status code is: ', exception.status_code)
```
or
```
    print('The status code is: ', exception.code())
```

IOW, I don't think the user can continue to just use `exception.code`, can they???

Reply:
Fortunately that's not the case. In short, the `@property` decorator creates a valid attribute (well only with a setter by default), so it can be used as before. See the example below;
```
>>> ex = ApiException(404, message="Not found...")
>>> ex.code
404
>>> ex.status_code
404
```
The only difference is, when calling the `code` property, a deprecation message will be printed to the `stderr` at warning level.

Reply:
Aha! I didn't notice the `@property` decorator, and I probably wouldn't have known what that was for even if I did :joy:.